productinfo/ec2/productinfo_ec2.go

```diff
@@ -159,6 +176,148 @@ func (e *Ec2Infoer) GetProducts(regionId string, attrKey string, attrValue produ
 	return vms, nil
 }
 
+type priceData struct {
+	awsData aws.JSONValue
+	attrMap map[string]interface{}
+}
+
+func newPriceData(prData aws.JSONValue) (*priceData, error) {
+	pd := priceData{awsData: prData}
+
+	// get the attributes map
+	productMap, err := getMapForKey("product", pd.awsData)
+	if err != nil {
+		return nil, err
+	}
+
+	attrMap, err := getMapForKey("attributes", productMap)
+	if err != nil {
+		return nil, err
+	}
+
+	pd.attrMap = attrMap
+
+	return &pd, nil
+}
+
+func (pd *priceData) GetInstanceType() (string, error) {
+
+	instanceType, ok := pd.attrMap["instanceType"]
+	if !ok {
+		return "", errors.New("could not get instance type")
+	}
+
+	instanceTypeStr, ok := instanceType.(string)
+	if !ok {
+		return "", errors.New("could not cast instance type to string")
+	}
+
+	return instanceTypeStr, nil
+}
+
+func (pd *priceData) GetVcpu() (string, error) {
+
+	vcpu, ok := pd.attrMap["vcpu"]
+
+	if !ok {
+		return "", errors.New("could not get vcpu")
+	}
+
+	vcpuStr, ok := vcpu.(string)
+	if !ok {
+		return "", errors.New("could not cast vcpu to string")
+	}
+
+	return vcpuStr, nil
+}
+
+func (pd *priceData) GetMem() (string, error) {
+
+	mem, ok := pd.attrMap[productinfo.Memory]
+
+	if !ok {
+		return "", errors.New("could not get memory")
+	}
+
+	memStr, ok := mem.(string)
+	if !ok {
+		return "", errors.New("could not cast memory to string")
+	}
+
+	return memStr, nil
+}
+
+func (pd *priceData) GetGpu() (string, error) {
+
+	gpu, ok := pd.attrMap["gpu"]
+
+	if !ok {
+		return "", errors.New("could not get gpu")
+	}
+
+	gpuStr, ok := gpu.(string)
+	if !ok {
+		return "", errors.New("could not cast gpu to string")
+	}
+
+	return gpuStr, nil
+}
```

Comment:
maybe it's just me, but these blocks could be shorter, something like this (or at least remove some empty lines please):
```
if vcpu, ok := pd.attrMap["vcpu"]; !ok {
	   	return "", errors.New("could not get vcpu")
	} else if vcpuStr, ok := vcpu.(string); !ok {
		return "", errors.New("could not cast vcpu to string")
	} else {
		return vcpuStr, nil
	}
```
also, the above 4 functions are copy-pasted, and can be substituted by only 1 function that gets the attribute name as a parameter